### PR TITLE
refactor: replace Expr with datafusion::Expr

### PR DESCRIPTION
**Pull request by realtaobo**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
as title, part of #3884 

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/metric-engine/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ common-recordbatch.workspace = true
 common-telemetry.workspace = true
 common-time.workspace = true
 datafusion.workspace = true
+datafusion-expr.workspace = true
```
Comment (evenyag) [inline src/metric-engine/Cargo.toml:22]: We already import datafusion so we should be able to import the Expr by `use datafusion::logical_expr::Expr`.

File: src/catalog/src/information_schema/predicate.rs [hunk 2]
```diff
@@ -14,10 +14,10 @@
 
 use arrow::array::StringArray;
 use arrow::compute::kernels::comparison;
-use common_query::logical_plan::DfExpr;
 use datafusion::common::ScalarValue;
 use datafusion::logical_expr::expr::Like;
 use datafusion::logical_expr::Operator;
+use datafusion::prelude::Expr;
```
Comment (evenyag) [inline src/catalog/src/information_schema/predicate.rs:20]: non-blocking: Maybe import Expr from logical_expr.
```suggestion
use datafusion::logical_expr::{Expr, Operator};
```

File: src/metric-engine/src/engine/read.rs [hunk 3]
```diff
@@ -15,10 +15,10 @@
 use std::sync::Arc;
 
 use api::v1::SemanticType;
-use common_query::logical_plan::Expr;
 use common_recordbatch::SendableRecordBatchStream;
 use common_telemetry::{error, info, tracing};
 use datafusion::logical_expr;
+use datafusion_expr::expr::Expr;
```
Comment (evenyag) [inline src/metric-engine/src/engine/read.rs:21]:
```suggestion
use datafusion::logical_expr::{self, Expr};
```

**Review by evenyag — APPROVED** [review]
LGTM. Thanks! @realtaobo I have linked this PR to the issue's task list.

**Review by waynexia — APPROVED** [review]
There are still many alias `DfExpr`, but this PR is good to go

**Pull request merged**